feat(catalog): add mongodb native connection details (#2052)
mia-platform/documentation: docs/data_catalog/frontend/data_catalog_connections.mdx

  </div>
</div>

:::info
Since __MongoDB__ is a NoSQL database there is no concept of schema for a collection. However when a [JSON Schema validation](https://www.mongodb.com/docs/manual/core/schema-validation/) is available
it may make sense to detect fields and types for a collection.

Types from `MongoDB` are directly picked from the [BSON Type specification](https://www.mongodb.com/docs/manual/reference/operator/query/jsonSchema/#mongodb-query-op.-jsonSchema). 
During the retrieval phase, the agent follows these rules to parse the validation object:

- when BSON and JSON schemas overlap (please refer to [this table](https://docs.rs/jsonutil/latest/jsonutil/index.html) for a comparison list used by the inner library), then the JSON schema is included;
- when the form of the validation is, at first level, written as:

  ```json
  {
    "type": "object",
    "properties": {
      // a list of fields
    }
  }
  ```

  then fields are extracted, since it is possible to determine column assets for each document in the collection;
- if the validation schema contains `anyOf`, `allOf`, or `oneOf` at the first level, the schema **won't be parsed**.

:::

> Other than the connection settings, it's possible to also tune both databases and collections retrieval.

Reviewer: ```suggestion
In addition to the connection settings, it is also possible to tune both databases and collections retrieval.
```
